_episodes/l3_03_test_fixture.md

```diff
@@ -0,0 +1,505 @@
+---
+title: "Introduction to Test Fixtures using GoogleTest"
+teaching: TBC
+exercises: TBC
+questions:
+- "What is a test fixture?"
+- "How to create your own test fixture using GoogleTest?"
+- "What does Setup and Teardown mean while using test fixtures?"
+objectives:
+- "Understand the basics of test fixture."
+- "Understand how to create test fixture using GoogleTest."
+- "Understand the advantages of test fixtures."
+- "Understand the need of setup and teardown functions."
+- "Understand how to write your own setup and teardown function."
+- "Understand how to run a subset of tests."
+keypoints:
+- "We learnt about test fixtures in GoogleTest."
+- "We learnt how to create our own test fixtures."
+- "We learnt about the setup and teardown functions."
+- "We learnt the advantages of test fixtures."
+- "We learnt how to run a subset of tests in GoogleTest."
+---
+## 1. Brief Introduction of Test Fixtures
+
+A test fixture, in the context of unit testing, refers to the preparation and configuration needed to run a set of test cases. It includes the setup of the test environment, the creation of necessary objects or resources, and the cleanup steps after the tests are executed. Test fixtures help ensure that the tests are performed in a controlled and consistent environment, providing reliable and reproducible results.
+
+In GoogleTest, a test fixture is implemented using a test fixture class. This class serves as a container for shared setup and cleanup logic, as well as any shared objects or resources required by the test cases within the fixture.
+
+### 1.1 When is a Test Fixture Needed?
+
+A test fixture is typically used in the following scenarios:
+
+1. **Shared Setup and Teardown**: When multiple test cases require a common set of setup or teardown (which means cleaning in general and is the opposite of setup) steps, a test fixture is beneficial. Instead of duplicating the setup and teardown code in each test case, we can define it once in the test fixture and reuse it across all the tests.
```

Comment:
I think this point might be confusing since setup and teardown have not been formally introduced yet

Reply:
This was suggested to be rephrased by one of the reviewers and the suggestion was accepted. I feel that it is an important point which emphasize the need of test fixtures. Please see commit 09080a880c79ceb18394459e56d2075784745ec3